Sources/TensorFlow/Core/Utilities.swift

```diff
@@ -167,3 +167,157 @@ extension UnsafeMutablePointer where Pointee == CTensorHandle? {
+                max - 1 - (msgLength % blockSize) :
+                blockSize + max - 1 - (msgLength % blockSize))
+
+        // Step 2: Append the message length as a 64-bit representation of `lengthInBits`.
+        accumulated += lengthBytes
+
+        // Step 3: Process the array bytes.
+        var accumulatedHash = SIMD8<UInt64>(
+            0x6a09e667f3bcc908, 0xbb67ae8584caa73b, 0x3c6ef372fe94f82b, 0xa54ff53a5f1d36f1, 
+            0x510e527fade682d1, 0x9b05688c2b3e6c1f, 0x1f83d9abfb41bd6b, 0x5be0cd19137e2179)
+        var index = 0
+        while index < accumulated.count {
+            let chunk = accumulated[index..<(index + blockSize)]
+            index += blockSize
+            
+            // Break chunk into sixteen 64-bit words M[j], 0 ≤ j ≤ 15, in big-endian format.
+            // Extend the sixteen 64-bit words into eighty 64-bit words:
+            var M = [UInt64](repeating: 0, count: k.count)
+            for x in k.indices {
+                switch x {
+                case 0...15:
+                    let start = chunk.startIndex.advanced(by: x * 8)
+                    M[x] = UInt64(bytes: chunk, startingAt: start)
+                    break
+                default:
+                    let s0 = M[x - 15].rotate(rightBy: 1) ^ 
+                        M[x - 15].rotate(rightBy: 8) ^ 
+                        (M[x - 15] >> 7)
+                    let s1 = M[x - 2].rotate(rightBy: 19) ^ 
+                        M[x - 2].rotate(rightBy: 61) ^ 
+                        (M[x - 2] >> 6)
+                    M[x] = M[x - 16] &+ s0 &+ M[x - 7] &+ s1
+                    break
+                }
+            }
+
+            var hashCopy = accumulatedHash
+            for j in 0..<k.count {
+                let s0 = hashCopy[0].rotate(rightBy: 28) ^
+                    hashCopy[0].rotate(rightBy: 34) ^
+                    hashCopy[0].rotate(rightBy: 39)
+                let maj = (hashCopy[0] & hashCopy[1]) ^ 
+                    (hashCopy[0] & hashCopy[2]) ^
+                    (hashCopy[1] & hashCopy[2])
+                let t2 = s0 &+ maj
+                let s1 = hashCopy[4].rotate(rightBy: 14) ^
+                    hashCopy[4].rotate(rightBy: 18) ^
+                    hashCopy[4].rotate(rightBy: 41)
+                let ch = (hashCopy[4] & hashCopy[5]) ^ ((~hashCopy[4]) & hashCopy[6])
+                let t1 = hashCopy[7] &+ s1 &+ ch &+ k[j] &+ UInt64(M[j])
+                hashCopy[7] = hashCopy[6]
+                hashCopy[6] = hashCopy[5]
+                hashCopy[5] = hashCopy[4]
+                hashCopy[4] = hashCopy[3] &+ t1
+                hashCopy[3] = hashCopy[2]
+                hashCopy[2] = hashCopy[1]
+                hashCopy[1] = hashCopy[0]
+                hashCopy[0] = t1 &+ t2
+            }
+            accumulatedHash &+= hashCopy
+        }
+
+        // Step 4: Return the computed hash.
+        var result = [UInt8](repeating: 0, count: digestLength)
```

Comment:
Should we use `SIMD64<UInt8>` instead?

Reply:
That way, the following loop can also be vectorized.

Reply:
I use `[UInt8]` because we also define it over `[UInt8]` and we only afterwards break it down to a pair of `Int32`s. Do you think using `SIMD` would be useful here?

Reply:
If it has a fixed size, it'd be really great to use SIMD.